sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1643,12 +1643,16 @@ object SQLConf {
   val STORE_ASSIGNMENT_POLICY =
     buildConf("spark.sql.storeAssignmentPolicy")
       .doc("When inserting a value into a column with different data type, Spark will perform " +
-        "type coercion. Currently we support 3 policies for the type coercion rules: ansi, " +
-        "legacy and strict. With ansi policy, Spark performs the type coercion as per ANSI SQL. " +
-        "With legacy policy, Spark allows casting any value to any data type. " +
-        "The legacy policy is the only behavior in Spark 2.x and it is compatible with Hive. " +
+        "type coercion. Currently, we support 3 policies for the type coercion rules: ANSI, " +
+        "legacy and strict. With ANSI policy, Spark performs the type coercion as per ANSI SQL. " +
+        "In practice, it follows PostgreSQL. It disallows certain unreasonable type conversions " +
```

Comment:
Out of curiosity, should we mention PostgreSQL? For instance, if some behaviours from PostgreSQL don't comply ANSI, and don't quite make sense to Spark community, I think we might decide to don't follow just like we didn't decided to don't add some PostgreSQL corresponding features.

Reply:
This is because the ANSI store assignment rules are high level specified.
Take the `string` -> `int` as an example, it doesn't explicit specifying whether string columns can be converted to integer columns. I tried Mysql/Oracle and the conversion is allowed, while PostgreSQL disallows it. 
As we are porting tests from PostgreSQL (SPARK-27763), Spark tends to follow PostgreSQL in behaviors. On the table insertion, I think the rules of PostgreSQL are more reasonable. For more details see the docs summarized by @maropu  https://docs.google.com/spreadsheets/u/1/d/1I4pl1TIe8u0G4sGSc7OCItySfvK0pOTkU4IPw5Z0Qck/edit?usp=sharing .

So I think we should point out that we are following PostgreSQL particularly.

Reply:
I think @HyukjinKwon does have a point here. We don't promise that Spark always follows Postgres.

How about `In practice, the behavior is mostly the same as PostgreSQL.`

Reply:
@cloud-fan Thanks, I have updated the doc. 
If @HyukjinKwon still feel confusing on the doc, I am OK to just remove `In practice, the behavior is mostly the same as PostgreSQL.`.